dash/development/_jl_components_generation.py

```diff
@@ -0,0 +1,523 @@
+    """
+    filtered_props = copy.deepcopy(props)
+
+    for arg_name, arg in list(filtered_props.items()):
+        if "type" not in arg and "flowType" not in arg:
+            filtered_props.pop(arg_name)
+            continue
+
+        # Filter out functions and instances --
+        if "type" in arg:  # These come from PropTypes
+            arg_type = arg["type"]["name"]
+            if arg_type in {"func", "symbol", "instanceOf"}:
+                filtered_props.pop(arg_name)
+        elif "flowType" in arg:  # These come from Flow & handled differently
+            arg_type_name = arg["flowType"]["name"]
+            if arg_type_name == "signature":
+                # This does the same as the PropTypes filter above, but "func"
+                # is under "type" if "name" is "signature" vs just in "name"
+                if "type" not in arg["flowType"] or arg["flowType"]["type"] != "object":
+                    filtered_props.pop(arg_name)
+        else:
+            raise ValueError
+
+    return filtered_props
+
+
+def get_jl_type(type_object):
+    """
+    Convert JS types to Julia types for the component definition
+    Parameters
+    ----------
+    type_object: dict
+        react-docgen-generated prop type dictionary
+    Returns
+    -------
+    str
+        Julia type string
+    """
+    js_type_name = type_object["name"]
+    js_to_jl_types = get_jl_prop_types(type_object=type_object)
+    if js_type_name in js_to_jl_types:
+        prop_type = js_to_jl_types[js_type_name]()
+        return prop_type
+    return ""
+
+
+def print_jl_type(typedata):
+    typestring = get_jl_type(typedata).capitalize()
+    if typestring:
+        typestring += ". "
+    return typestring
+
+
+def create_docstring_jl(component_name, props, description):
+    """Create the Dash component docstring.
+    Parameters
+    ----------
+    component_name: str
+        Component name
+    props: dict
+        Dictionary with {propName: propMetadata} structure
+    description: str
+        Component description
+    Returns
+    -------
+    str
+        Dash component docstring
+    """
+    # Ensure props are ordered with children first
+    props = reorder_props(props=props)
+
+    return (
+        """A{n} {name} component.\n{description}
+Keyword arguments:\n{args}"""
+    ).format(
+        n="n" if component_name[0].lower() in ["a", "e", "i", "o", "u"] else "",
+        name=component_name,
+        description=description,
+        args="\n".join(
+            create_prop_docstring_jl(
+                prop_name=p,
+                type_object=prop["type"] if "type" in prop else prop["flowType"],
+                required=prop["required"],
+                description=prop["description"],
+                indent_num=0,
+            )
+            for p, prop in list(filter_props(props).items())
+        ),
+    )
+
+
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-arguments
```

Comment:
🔪 - both of them in fact 😅 Doesn't look necessary in the Python or R variants anymore either...

Reply:
fixed in 5b2edcdc